Source/WTF/wtf/SIMDHelpers.h

```diff
@@ -116,13 +166,61 @@ ALWAYS_INLINE std::optional<uint8_t> findFirstNonZeroIndex(simde_uint16x8_t valu
         return std::nullopt;
     return std::countr_zero(mask) >> 1;
 #else
+    //  value                     |c|c|c|C|c|C|c|c| (c is character, C is matching character)
+    //  eq with charactersVector  |0|0|0|X|0|X|0|0| (X is all-1)
+    //  Reduce to uint8x8_t       |0|0|0|X|0|X|0|0| => reinterpret it as uint64_t. If it is non-zero, matching character exists.
     if (!isNonZero(value))
         return std::nullopt;
     constexpr simde_uint16x8_t indexMask { 0, 1, 2, 3, 4, 5, 6, 7 };
+    // Found elements are all-1 and the other elements are 0. But it is possible that this vector
+    // includes multiple found characters. We perform [0, 1, 2, 3, 4, 5, 6, 7] OR-NOT with this mask,
+    // to assign the index to found characters.
+    // Find the smallest value. Because of [0, 1, 2, 3, 4, 5, 6, 7], the value should be index in this vector.
+    // If the index less than length, it is within the requested pointer. Otherwise, nullptr.
+    //
+    // Example
+    //     mask        |0|0|0|X|0|X|0|0| (X is all-one)
+    //     not-mask    |X|X|X|0|X|0|X|X|
```

Comment:
Let's use the term "value" instead of "mask" in the comment lines so that it matches the incoming argument name.  Otherwise, it can be confusing.

Reply:
Changed.